crates/rolldown_plugin_module_federation/src/lib.rs

```diff
@@ -1,23 +1,89 @@
-use std::borrow::Cow;
+use std::{
+  borrow::Cow,
+  sync::atomic::{AtomicBool, Ordering},
+};
 
 mod option;
 pub use option::{ModuleFederationPluginOption, Remote, Shared};
-use rolldown_plugin::Plugin;
+use rolldown_common::EmittedChunk;
+use rolldown_plugin::{HookResolveIdReturn, Plugin};
+use rolldown_utils::concat_string;
+
+const REMOTE_ENTRY: &str = "mf:remote-entry.js";
 
 #[derive(Debug)]
 pub struct ModuleFederationPlugin {
   #[allow(dead_code)]
   options: ModuleFederationPluginOption,
+  remote_entry_added: AtomicBool,
 }
 
 impl ModuleFederationPlugin {
   pub fn new(options: ModuleFederationPluginOption) -> Self {
-    Self { options }
+    Self { options, remote_entry_added: AtomicBool::default() }
   }
 }
 
 impl Plugin for ModuleFederationPlugin {
   fn name(&self) -> Cow<'static, str> {
     Cow::Borrowed("builtin:module-federation")
   }
+
+  async fn resolve_id(
+    &self,
+    ctx: &rolldown_plugin::PluginContext,
+    args: &rolldown_plugin::HookResolveIdArgs<'_>,
+  ) -> HookResolveIdReturn {
+    if !self.remote_entry_added.load(Ordering::Relaxed) {
```

Comment:
Here the `remote_entry_added` could be remove if support emit chunk at buildStart hook.